### PR TITLE
Bach: support athena date

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Had to open issue for date formatting (not that easy as expected):

https://github.com/objectiv/objectiv-analytics/issues/1154

### CODE REVIEW

File: bach/tests/functional/bach/test_series_date.py [hunk 1]
```diff
@@ -30,53 +31,25 @@ def test_date_comparator(asstring: bool, engine):
...
-        expected_data=[
-            [1, date(2021, 5, 3)],
-            [2, date(2021, 5, 4,)],
-            [4, date(2022, 5, 3)]
-        ]
-    )
+    mt['lte'] = mt['date'] <= dt
+    mt['lt'] = mt['date'] < dt
 
+    result = mt[['eq', 'gte', 'gt', 'lte', 'lt']]
```
Comment (jansentom) [inline bach/tests/functional/bach/test_series_date.py:45]: Nice speedup! gte and lte are not tested very well though (all True).

File: bach/bach/series/series_datetime.py [hunk 2]
```diff
@@ -360,7 +360,9 @@ def dtype_to_expression(cls, dialect: Dialect, source_dtype: str, expression: Ex
     def _comparator_operation(self, other, comparator,
                               other_dtypes=('timestamp', 'date', 'time', 'string'),
                               strict_other_dtypes=tuple()) -> 'SeriesBoolean':
-        return super()._comparator_operation(other, comparator, other_dtypes, strict_other_dtypes)
+        return super()._comparator_operation(
+            other, comparator, other_dtypes, strict_other_dtypes=tuple(self.dtype)
```
Comment (jansentom) [inline bach/bach/series/series_datetime.py:364]: are strict other dtypes requiredfor all data stores now? Causes an extra cast now if it's not required.

**Review by jansentom — APPROVED** [review]
Okay in principle, but spotted one weak test.

File: bach/bach/series/series_datetime.py (hunk 2 above)
Comment (KathiaBarahona) [reply bach/bach/series/series_datetime.py:364]: yes, the same error was raised for most of series date comparisons for athena

Comment (KathiaBarahona) [reply bach/bach/series/series_datetime.py:364]: Improved here: https://github.com/objectiv/objectiv-analytics/pull/1155/commits/19c14dde93c97e438b68d9f7d54354eba41facf0

**Pull request merged**